packages/flutter/test/cupertino/segmented_control_test.dart

```diff
@@ -119,6 +119,56 @@ void main() {
     }
   });
 
+  testWidgets('Default padding is set', (WidgetTester tester) async {
+    final Map<int, Widget> children = <int, Widget>{};
+    children[0] = const Text('Child 1');
+    children[1] = const Text('Child 2');
+
+    await tester.pumpWidget(
+      StatefulBuilder(
+        builder: (BuildContext context, StateSetter setState) {
+          return boilerplate(
+            child: CupertinoSegmentedControl<int>(
+              key: const ValueKey<String>('Segmented Control'),
+              children: children,
+              onValueChanged: (int newValue) {},
+              groupValue: 1,
+            ),
+          );
+        },
+      ),
+    );
+
+    expect(find.byType(Padding), findsOneWidget);
```

Comment:
Inspecting `Padding` directly to see if the parameter is set correctly seems a bit implementation dependent, IMO it's better to verify the expected behavior.

I'd prefer to verify that the bounding `Rect` of child widgets is indeed inset by `padding` amount, i.e. verify the top left corner of "Child 1" is `Offset(16, 0)` to the top left corner of `CupertinoSegmentedControl<int>` etc..

Reply:
How should I implement this test? coordinates_test.dart is what I am looking at at this point. 

```
    final RenderBox boxA = tester.renderObject(find.byKey(keyChild1));
    expect(boxA.globalToLocal(Offset.zero), equals(const Offset(0.0, 286.0)));```

How can I check what the position is between 2 widgets?

Reply:
The [WidgetTester](https://api.flutter.dev/flutter/flutter_test/WidgetTester-class.html) class offers methods to get the key coordinates of a widget, e.g. `getBottomLeft`

Reply:
e.g.
```dart
  testWidgets('Padding works', (WidgetTester tester) async {
    const Key key = Key('Container');

    final Map<int, Widget> children = <int, Widget>{};
    children[0] = const SizedBox(
      height: double.infinity,
      child: Text('Child 1'),
    ) ;
    children[1] = const SizedBox(
      height: double.infinity,
      child: Text('Child 2'),
    ) ;

    Future<void> verifyPadding({ EdgeInsets padding }) async {
      final EdgeInsets effectivePadding = padding ?? const EdgeInsets.symmetric(horizontal: 16);
      final Rect segmentedControlRect = tester.getRect(find.byKey(key));
      expect(
        tester.getTopLeft(find.byWidget(children[0])),
        segmentedControlRect.topLeft.translate(
          effectivePadding.topLeft.dx,
          effectivePadding.topLeft.dy,
        )
      );
      expect(
        tester.getBottomLeft(find.byWidget(children[0])),
        segmentedControlRect.bottomLeft.translate(
          effectivePadding.bottomLeft.dx,
          effectivePadding.bottomLeft.dy,
        ),
      );

      expect(
        tester.getTopRight(find.byWidget(children[1])),
        segmentedControlRect.topRight.translate(
          effectivePadding.topRight.dx,
          effectivePadding.topRight.dy,
        ),
      );
      expect(
        tester.getBottomRight(find.byWidget(children[1])),
        segmentedControlRect.bottomRight.translate(
          effectivePadding.bottomRight.dx,
          effectivePadding.bottomRight.dy,
        ),
      );
    }

    await tester.pumpWidget(
      boilerplate(
        child: CupertinoSegmentedControl<int>(
          key: key,
          children: children,
          onValueChanged: (int newValue) { },
        ),
      )
    );

    // Default padding works.
    await verifyPadding();

    // Switch to Child 2 padding should remain the same.
    await tester.tap(find.text('Child 2'));
    await tester.pumpAndSettle();

    await verifyPadding();

    await tester.pumpWidget(
      boilerplate(
        child: CupertinoSegmentedControl<int>(
          key: key,
          padding: const EdgeInsets.fromLTRB(1, 3, 5, 7),
          children: children,
          onValueChanged: (int newValue) { },
        ),
      )
    );

    // Custom padding works.
    await verifyPadding(padding: const EdgeInsets.fromLTRB(1, 3, 5, 7));

    // Switch back to Child 1 padding should remain the same.
    await tester.tap(find.text('Child 1'));
    await tester.pumpAndSettle();

    await verifyPadding(padding: const EdgeInsets.fromLTRB(1, 3, 5, 7));
  });
```